File: PDOO/Material Complementario/Tema7/ejemplosTema7/ImplementacionEjemplosTema7/src/ejemplodsecuencia/ConciertoBenefico.java

package ejemplodsecuencia;

//solo se implementa aquello directamente relacionado con el diagrama de secuencia del ejemplo

public class ConciertoBenefico {
    private Sala lugarCelebracion;
    
    public String comprarEntradas(String nombreCliente, Date fecha, int numeroEntradas){
        
        Sala s= getLugarCelebracion(); //envío de mensaje 2.1
        
        int nReservas= s.obtenerNumeroReservas(fecha); //envío de mensaje 2.2
        
        int aforo= s.getAforo(); //envío de mensaje 2.3
        
        if (nReservas+numeroEntradas<aforo){ //inicio bloque alt
            
            s.reservaPlazas(nombreCliente, fecha, numeroEntradas); //envío de mensaje 2.4
        
            return "Compra efectuada: las reservas solicitadas están hechas a su nombre";}
            
        else  //segunda parte del bloque alt
            
             return "Compra denegada: no hay aforo suficiente";
        
        
        //ver 2.4.1 y 2.4.2 en la clase Sala, método reservaPlazas
            
    }   
       
        
   
    public Sala getLugarCelebracion{
        return lugarCelebracion;
}
    
}

File: PDOO/Material Complementario/Tema7/ejemplosTema7/ImplementacionEjemplosTema7Ruby/lib/ejemplodsecuencia/concierto_benefico.rb
class ConciertoBenefico
  
  attr_reader :lugar_celebracion
  
  
  def initialize(lc)
    @lugar_celebracion= lc
  end
  
    
 def comprar_entradas(nombre_cliente, fecha, numero_entradas){
        
        s= @lugar_celebracion #envío de mensaje 2.1
        
        nReservas= s.obtener_numero_reservas(fecha) #envío de mensaje 2.2
        
        aforo= s.aforo #envío de mensaje 2.3
        
        if (nReservas+numero_entradas<aforo){ #inicio bloque alt
            
            s.reserva_plazas(nombre_cliente, fecha, numero_entradas) #envío de mensaje 2.4
        
            return 'Compra efectuada: las reservas solicitadas están hechas a su nombre'
            
        else  #{}segunda parte del bloque alt
            
             return 'Compra denegada: no hay aforo suficiente';
        
        
        # ver 2.4.1 y 2.4.2 en la clase Sala, método reserva_plazas
            
    }   
       
        
   

    
end
